### PR TITLE
Fixes #14523 - Prevent table from disappearing on sort

### PR DESCRIPTION
**Pull request by xprazak2**
I found this in multiple tables. Needs [this PR](https://github.com/Katello/bastion/pull/107) for Bastion to work in all cases.

### CODE REVIEW

**Comment by waldenraines** [comment]
I think the commit message should be changed to something like "Prevent table from disappearing on sort".

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/subscriptions/subscriptions.controller.js [hunk 1]
```diff
@@ -39,11 +39,6 @@ angular.module('Bastion.subscriptions').controller('SubscriptionsController',
             $scope.transitionTo('subscriptions.index');
         };
 
-        $scope.groupedSubscriptions = {};
-        $scope.$watch('table.rows', function (rows) {
-            $scope.groupedSubscriptions = SubscriptionsHelper.groupByProductName(rows);
-        });
-
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/subscriptions/subscriptions.controller.js]: @thomasmckay do you have any thoughts on removing the grouping of subscriptions?

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/subscriptions/views/subscriptions-table-collapsed.html [hunk 2]
```diff
@@ -6,14 +6,13 @@
...
-    <tr bst-table-row ng-repeat-start="(name, subscriptions) in groupedSubscriptions">
+    <tr bst-table-row ng-repeat-start="subscription in table.rows">
       <td bst-table-cell>
-        <b>{{ name }}</b>
+        <b>{{ subscription.name }}</b>
       </td>
     </tr>
     <tr bst-table-row
         ng-repeat-end
-        ng-repeat="subscription in subscriptions"
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/subscriptions/views/subscriptions-table-collapsed.html]: @thomasmckay

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/subscriptions/subscriptions.controller.js (hunk 1 above)
Comment (thomasmckay) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/subscriptions/subscriptions.controller.js]: Why would grouping be removed? It's kind of a key feature of subscription viewing.

Comment (xprazak2) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/subscriptions/subscriptions.controller.js]: The page looks the same as before, I just think there is a simpler way to achieve it.

Comment (thomasmckay) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/subscriptions/subscriptions.controller.js]: Note difference when subscriptions are grouped by name. It is an important feature and quite common experience for customers. Let me know if you have other questions. Thanks!

subscriptions master

subscriptions

**Comment by xprazak2** [comment]
I see now. I will leave the grouping as is.

**Comment by waldenraines** [comment]
ACK, thanks @xprazak2

**Pull request merged**